src/navigator/QRNavigator.tsx

```diff
@@ -95,21 +96,23 @@ function AnimatedScannerScene({ route, position, ...props }: AnimatedScannerScen
   const { scanIsForSecureSend, isOutgoingPaymentRequest, transactionData, requesterAddress } =
     route.params || {}
 
-  const onBarCodeDetected = memoize(
```

Comment:
the `memoize` was causing me problems, and i also don't think it adds value here? why do we need to memoize this function when it's not doing any heavy computations and in theory we only want it to execute once per `qrCode.data` anyway?

Reply:
This was how it initially avoided this bug.
But it got incorrectly refactored: [packages/mobile/src/qrcode/QRScanner.tsx](https://github.com/valora-inc/wallet/commit/6e3de6570f42c9917226d3545561fcf914f59a7e#diff-0f9c2a2d085bf912dedd013ea8ea93a98046538b740e6d32176330fe643cd2c6)
See the missing `useCallback` which made it call `memoize` over and over on each render.

Clever code was too clever :D

`memoize` was first added [here](https://github.com/valora-inc/wallet/commit/437a93854a58f5e9d98ffc784663eacb31dc277a) and then I'm guilty of continuing to use it [here](https://github.com/valora-inc/wallet/commit/093734dda4bb7ba023d2013a342b7d2ca1bebfb0#diff-0f9c2a2d085bf912dedd013ea8ea93a98046538b740e6d32176330fe643cd2c6).